src/core/vector/qgsvectorlayerutils.h

```diff
@@ -382,9 +382,17 @@ class CORE_EXPORT QgsVectorLayerUtils
      * Assumes that the user has organized the data with the more "interesting" field
      * names first. As such, "name" would be selected before "oldname", "othername", etc.
      *
+     * Since QGIS 3.22, an optional \a foundFriendly parameter can be used
+     * to determine whether the returned field name  is a friendly identifier or
+     * had to fallback to the first available field.
+     *
+     * \note The foundFriendly parameter is available in Python bindings by using the guessFriendlyIdentifierFieldV2 function
      * \since QGIS 3.18
      */
+#ifdef SIP_RUN
     static QString guessFriendlyIdentifierField( const QgsFields &fields );
+#endif
+    static QString guessFriendlyIdentifierField( const QgsFields &fields, bool *foundFriendly = nullptr ) SIP_PYNAME( guessFriendlyIdentifierFieldV2 );
```

Comment:
```suggestion
    static QString guessFriendlyIdentifierField( const QgsFields &fields, bool *foundFriendly SIP_OUT = nullptr ) SIP_PYNAME( guessFriendlyIdentifierFieldV2 );
```

But you'll also need to fix the docs here, because SIP_OUT requires a full /params listing and /returns annotations

Reply:
@nyalldawson , ok, will do.